File: Data Structures & Algorithms/products-of-array-discluding-self/submission-2.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        // We want to 1, 2, 4, 6 -> 48, 24, 12, 8
        // The tricky part is we need to get the previous number to multiply
        // How about if we go forward -> 1, 1, 2, 8
        //  1, 2, 4, 6  now we do the same thing right side
        //  48, 24, 6, 1
        // now we just multiply with our other array
        // 48, 24, 12, 8

        vector<int> ans;
        ans.resize(nums.size(), 1);
        for (int i = 1; i < nums.size(); ++i) { // we skip the first element since that will be one
            ans[i] = ans[i - 1] * nums[i - 1];
        }

        int right = 1;
        for (int i = nums.size() - 1; i >= 0; --i) {
            ans[i] *= right;
            right *= nums[i];
        }

        return ans;
    }
};

File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.cs
public class Solution 
{
    public int[] TwoSum(int[] nums, int target) 
    {
        int left = 0;
        int right = nums.Length - 1;

        while (left <= right)
        {
            int current = nums[left] + nums[right];

            if (current == target)
            {
                return new int[] {left + 1, right + 1};
            }
            else if (current < target) // we need to move our left pointer
            {
                left++;
            }
            else if (current > target)
            {
                right--;
            }
        }

        return new int[] {};
    }
}
